kernel/cbytegen.ml

```diff
@@ -501,6 +501,9 @@ let rec compile_lam env cenv lam sz cont =
       if Array.is_empty args then
         compile_fv_elem cenv (FVevar evk) sz cont
       else
+        (** Arguments are reversed in evar instances *)
+        let args = Array.copy args in
+        let () = Array.rev args in
```

Comment:
Ok for now, but can the order be fixed? Is there a reason for the discrepancy?

Reply:
You mean in the `Constr.Evar` node or somewhere else?

Reply:
Yeah, that's what I meant. I was referring to your comment.